mne/io/proj.py

```diff
@@ -108,7 +108,10 @@ def add_proj(self, projs, remove_existing=False, verbose=None):
     def add_eeg_average_proj(self):
         """Add an average EEG reference projector if one does not exist
         """
-        if _needs_eeg_average_ref_proj(self.info):
+        if self.info['custom_ref_applied']:
+            logger.warning('A custom reference was previously applied, '
+                           'average reference cannot be created.')
```

Comment:
It is perfectly valid to move from a custom referencing scheme to an average reference. There are even valid use cases for this.